modules/nf-commons/src/main/nextflow/util/CacheHelper.java

```diff
@@ -215,7 +216,7 @@ static private Hasher hashFile( Hasher hasher, Path path, HashMode mode ) {
         try {
             attrs = Files.readAttributes(path, BasicFileAttributes.class);
         }
-        catch(IOException e) {
+        catch(IOException | ProviderMismatchException e) {
             log.debug("Unable to get file attributes file: {} -- Cause: {}", FilesEx.toUriString(path), e.toString());
         }
 
```

Comment:
Thanks for sending this PR. It makes sense to capture this, tho maybe it could be useful to report a warning to the user because it's a quite unusual error and can signal something bad. What it's causing this error on your side?

Reply:
@pditommaso 

> What it's causing this error on your side?

I whish I knew :cry:

Reply:
@pditommaso so instead of `log.debug` , you wish I use` log.warn` for this kind of error ?

Reply:
Yes, `log.warn` even tho this can spam the program output. 

> I whish I knew

Usually, this error happens when composing two `Path` objects belonging to different file systems. For example: 

```
s3File = file('s3://data/there')
localFile = file('/data/here')
localFile.resolve(localFile)
```

Reply:
note: `file(something)` is just a wrapper around java `Paths.get(something)`

Reply:
@pditommaso 
> Usually, this error happens when composing two Path objects belonging to different file systems.

yes, it could be my problem but it's always the very same file among other files that are created the same way. For now, I solved this problem by making a copy of the file instead of of using the symllink created by NF...